### PR TITLE
add token 22 extension examples + seahorse examples

### PR DESCRIPTION
**Pull request by valentinmadrid**

### CODE REVIEW

**Comment by valentinmadrid** [comment]
@buffalojoec Double checked every program. You can merge.

File: basics/transfer-sol/seahorse/programs_py/seahorse.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@instruction
+def init_mock_account(signer: Signer, mock_account: Empty[MockAccount]):
+    account = mock_account.init(
+        payer = signer,
+        seeds = ['mock_account'],
+    )
+    account.mock_value = u8(0)
+
+@instruction
+def transfer(sender: Signer, recipient: MockAccount, amount: u64):
```
Comment (buffalojoec) [inline basics/transfer-sol/seahorse/programs_py/seahorse.py:22]: Can we use the same pattern & naming convention as the other two transfer examples? 

There's both a "transfer with CPI" and "transfer with program" instruction in each, and in both cases the account names are `payer` and `recipient`

File: basics/transfer-sol/seahorse/tests/seahorse.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { Program } from "@coral-xyz/anchor";
+import { Seahorse } from "../target/types/seahorse";
+
+describe("seahorse", () => {
+  // Configure the client to use the local cluster.
+  anchor.setProvider(anchor.AnchorProvider.env());
+
+  const program = anchor.workspace.Seahorse as Program<Seahorse>;
+
+  const mockAccount = anchor.web3.PublicKey.findProgramAddressSync(
```
Comment (buffalojoec) [inline basics/transfer-sol/seahorse/tests/seahorse.ts:11]: Same thing, account naming conventions should match the other transfer examples

File: tokens/token-2022/default-account-state/native/program/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "program"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+borsh = "0.9.3"
+borsh-derive = "0.9.1"
+solana-program = "1.10.26"
+spl-token = { version="3.3.0", features = [ "no-entrypoint" ] }
```
Comment (buffalojoec) [inline tokens/token-2022/default-account-state/native/program/Cargo.toml:10]: Probably don't need both dependencies (`spl-token` and `spl-token-2022`)

File: tokens/token-2022/default-account-state/native/program/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        ),
+        &[
+            mint_account.clone(),
+            payer.clone(),
+            system_program.clone(),
+            token_program.clone(),
+        ],
+    )?;
+
+    // Initialize the Default Account State as Frozen
```
Comment (buffalojoec) [inline tokens/token-2022/default-account-state/native/program/src/lib.rs:74]: Add a single line saying "you have to do this before initializing the mint" or "all extensions must be initialized before initializing the Token2022 mint" or something like that

File: tokens/token-2022/default-account-state/native/program/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        )?,
+        &[
+            mint_account.clone(),
+            mint_authority.clone(),
+            token_program.clone(),
+            rent.clone(),
+        ],
+    )?;
+
+    // Update the Default Account State to Initialized
```
Comment (buffalojoec) [inline tokens/token-2022/default-account-state/native/program/src/lib.rs:106]: I think you should probably document what's happening here in the README of `tokens/token-2022/default-account-state`.

A lot of people probably don't know what "initializing default account state" and "updating default account state" are, why you do them, when, etc.

File: tokens/token-2022/mint-close-authority/native/program/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "program"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+borsh = "0.9.3"
+borsh-derive = "0.9.1"
+solana-program = "1.10.26"
+spl-token = { version="3.3.0", features = [ "no-entrypoint" ] }
```
Comment (buffalojoec) [inline tokens/token-2022/mint-close-authority/native/program/Cargo.toml:10]: Same comment as above on double dependencies

**Pull request merged**